src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -1933,32 +1942,73 @@ TEST_CASE("txenvelope", "[tx][envelope]")
             });
         }
 
-        SECTION("remove signer, do something")
+        SECTION("reduce auth, do something")
         {
-            a.setOptions(setSigner(makeSigner(b, 1)) | setMasterWeight(1) |
-                         setLowThreshold(2) | setMedThreshold(2) |
-                         setHighThreshold(2));
+            a.setOptions(setSigner(makeSigner(b, 2)) | setMasterWeight(3) |
+                         setLowThreshold(1) | setMedThreshold(3) |
+                         setHighThreshold(5));
 
-            for_versions({1, 2, 3, 4, 5, 6, 8, 9}, *app, [&] {
-                auto tx = a.tx({setOptions(setSigner(makeSigner(b, 0))),
-                                setOptions(setHomeDomain("stellar.org"))});
-                tx->addSignature(b);
+            SECTION("single tx")
+            {
+                SECTION("valid")
+                {
+                for_versions({1, 2, 3, 4, 5, 6, 8, 9}, *app, [&] {
+                    auto tx = a.tx({setOptions(setSigner(makeSigner(b, 1))),
+                                    setOptions(setSigner(makeSigner(b, 2)))});
+                    tx->addSignature(b);
 
-                validateTxResults(
-                    tx, *app, {baseFee * 2, txSUCCESS},
-                    expectedResult(baseFee * 2, 2, txFAILED,
-                                   {SET_OPTIONS_SUCCESS, opBAD_AUTH}));
-            });
-            for_versions_from({7, 10}, *app, [&] {
-                auto tx = a.tx({setOptions(setSigner(makeSigner(b, 0))),
-                                setOptions(setHomeDomain("stellar.org"))});
-                tx->addSignature(b);
+                    validateTxResults(
+                        tx, *app, {baseFee * 2, txSUCCESS},
+                        expectedResult(baseFee * 2, 2, txFAILED,
+                                       {SET_OPTIONS_SUCCESS, opBAD_AUTH}));
+                });
+                for_versions_from({7, 10}, *app, [&] {
+                    auto tx = a.tx({setOptions(setSigner(makeSigner(b, 1))),
+                                    setOptions(setSigner(makeSigner(b, 2)))});
+                    tx->addSignature(b);
 
-                validateTxResults(
-                    tx, *app, {baseFee * 2, txSUCCESS},
-                    expectedResult(baseFee * 2, 2, txSUCCESS,
-                                   {SET_OPTIONS_SUCCESS, SET_OPTIONS_SUCCESS}));
-            });
+                    validateTxResults(tx, *app, {baseFee * 2, txSUCCESS},
+                                      expectedResult(baseFee * 2, 2, txSUCCESS,
+                                                     {SET_OPTIONS_SUCCESS,
+                                                      SET_OPTIONS_SUCCESS}));
+                });
+                }
+                SECTION("missing signature")
+                {
+                    for_versions_from(10, *app, [&] {
+                        auto tx =
+                            a.tx({payment(root, 1000),
```

Comment:
Nitpick: Formatting here is really weird, is this what clang-format produced?

Reply:
very strange indeed - yeah it was a clang-format issue